test/functional/apps/dashboard/_dashboard_grid.js

```diff
@@ -3,43 +3,125 @@ import expect from 'expect.js';
 
-    // Specific test after https://github.com/elastic/kibana/issues/14764 fix
-    it('Can move panel from bottom to top row', async function addVisualizations() {
-      await PageObjects.dashboard.clickNewDashboard();
-      await PageObjects.dashboard.addVisualizations([
-        PageObjects.dashboard.getTestVisualizationNames()[0],
-        PageObjects.dashboard.getTestVisualizationNames()[1],
-        PageObjects.dashboard.getTestVisualizationNames()[2],
-      ]);
+    describe('move panel', () => {
+      // Specific test after https://github.com/elastic/kibana/issues/14764 fix
+      it('Can move panel from bottom to top row', async () => {
+        await PageObjects.dashboard.clickNewDashboard();
+        await PageObjects.dashboard.addVisualizations([
+          PageObjects.dashboard.getTestVisualizationNames()[0],
+          PageObjects.dashboard.getTestVisualizationNames()[1],
+          PageObjects.dashboard.getTestVisualizationNames()[2],
+        ]);
+
+        const panels = await find.allByCssSelector('.panel-title');
+
+        const thirdPanel = panels[2];
+        const position1 = await thirdPanel.getPosition();
+
+        remote
+          .moveMouseTo(thirdPanel)
+          .pressMouseButton()
+          .moveMouseTo(null, -20, -400)
+          .releaseMouseButton();
+
+        const panelsMoved = await find.allByCssSelector('.panel-title');
+        const position2 = await panelsMoved[2].getPosition();
+
+        expect(position1.y).to.be.greaterThan(position2.y);
+      });
+    });
+
+    describe('resize panel', () => {
+
+      describe('input control panel', () => {
+
+        before(async () => {
+          await PageObjects.common.navigateToApp('dashboard');
+          await PageObjects.dashboard.clickNewDashboard();
+          await PageObjects.dashboard.addVisualizations(['Visualization InputControl']);
+        });
+
+        it('Should wrap controls in horizontal layout when width cannot fit all controls min width', async () => {
+          const controls = await find.allByCssSelector('.inputControlVis .kuiFlexItem');
+          expect(controls.length).to.equal(3);
+
+          const control0Position = await controls[0].getPosition();
+          const control1Position = await controls[1].getPosition();
+          const control2Position = await controls[2].getPosition();
+          expect(control0Position.y).to.equal(control1Position.y);
+          expect(control2Position.y).to.be.greaterThan(control0Position.y);
+        });
+
+        it('Should position controls in horizontal layout when panel is short and long', async () => {
+          const resizeIcons = await find.allByCssSelector('.react-resizable-handle');
+          expect(resizeIcons.length).to.equal(1);
+          remote
+            .moveMouseTo(resizeIcons[0])
+            .pressMouseButton()
+            .moveMouseTo(null, 300, 0)
+            .releaseMouseButton();
+          await PageObjects.common.sleep(500); //give time for vis to redraw
```

Comment:
`retry` does not work in this situation since the elements already exist and will be returned the first attempt. A sleep is needed before the `expect` to allow the elements to be re-drawn with the new state. I updated the commits around the sleep to reflect this.